apps/alpha-test-app/src/components/molecules/FormDemoMolecule.vue

```diff
@@ -0,0 +1,102 @@
+<script lang="ts" setup>
+import { TestInput } from "sit-onyx";
+import { ref, watch } from "vue";
+
+type Props = {
+  defaultInput: string;
+  requiredInput: string;
+  minlengthInput: string;
+  maxInput: number;
+  typeInput: string;
+  patternInput: string;
+};
+const props = defineProps<{
+  formData: Props;
+}>();
+
+const formState = ref<Props | undefined>();
+
+const emit = defineEmits<{
+  submit: [data: object];
+}>();
+
+const formElement = ref<HTMLFormElement | null>(null);
+
+const customErrorExample = ref("");
+
+const onSubmit = (event: Event) => {
+  event.preventDefault();
+  emit("submit", props);
+};
+
+const onPatternValidityChange = (state: ValidityState) => {
+  customErrorExample.value = state.patternMismatch
+    ? "Allows only lowercase characters or space"
+    : "";
+};
+
+watch(
+  props,
+  () => {
+    formState.value = { ...props.formData };
+  },
+  { immediate: true },
+);
+</script>
+
+<template>
+  <form v-if="formState" ref="formElement" class="demo" @submit="onSubmit($event)">
```

Comment:
```suggestion
  <form v-if="formState" ref="formElement" class="demo" @submit.prevent="emit('submit', props)">
```

(optional) Same behaviour + we could get rid of the `onSubmit` function :)